guides/documentations/essentials.md

```diff
@@ -0,0 +1,265 @@
+ComponentLinkDefinition.js,      // Class used for the definition of the link between Components.
+ParseError.js,                   // Class that represents a parsing error. Used by the DefaultParser as default Error.
+FileInformation.js,              // Class that represents the object to store all file information.
+FileInput.js,                    // Class that represents the object to store the file content.
+Tag.js,                          // Class that represents a tag in Leto-modelizer. Used by the DefaultConfiguration class.
+Variable.js,                     // Class that represents a variable in a file of a diagram. Used by the DefaultData class.
+```
+
+| Plugin lifecycle |
+| :--: |
+| <img height="400" src="../../documentations/lifecycle.svg"/> |
+
+This is the default lifecycle of plugin usage in Leto Modelizer.
+
+## Metadata
+
+To make your plugin work, we need to have an array of [ComponentDefinitions](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/models/ComponentDefinition.js).
+Component definitions are loaded from a metadata file. Each Component definition has:
+- A name
+- An icon
+- Attributes
+- ...
+
+| Left: Components in Leto-modelizer - Right: Related Metadata |
+| :--: |
+| <img height="400" src="../../documentations/components.png"/> |
+
+Attributes of a Component represent all the related properties that can/must be set.
+Link Definitions can also be loaded from the metadata file (for colored links, dashed links...).
+
+Once the Component definitions are loaded, components can be instantiated from the "components palette" in Leto-modelizer.
+The DefaultMetadata provides two methods:
+
+- `validate`: Validate the metadata.
+- `parse`: Set all component/link definitions in PluginData from your provided metadata.
+
+## Renderer / Parser
+
+| Parse and Render |
+| :--: |
+| <img height="400" src="../../documentations/parser_renderer.png"/> |
+
+### Renderer
+
+Serialize all the instantiated Components into files (`.tf` files for Terraform, `.yaml` files for Kubernetes, ...)
+It uses Jinja/Nunjucks templates.
+
+The DefaultRender provides these methods:
+
+- `render`: Transform all provided components in file inputs.
+- `renderFiles`: Maps all the given components into a file content and return all files.
+- `renderConfiguration`: Update configuration file content according to components data.
+
+### Parser
+
+Deserialize files to instantiate Components and their Attributes.
+It uses ANTLR or Lidy grammars.
+The DefaultParser provides these methods:
+
+- `parse`: Convert the content of files into Components.
+- `getModels`: Get the list of model paths from all parsable files.
+- `parseConfiguration`: Set configuration into Components.
+- `isParsable`: Indicate if this parser can parse this file. By default, this function return false only on null/undefined fileInformation.
+
+## Drawer
+
+The Drawer is used to define/override the way components are drawn on the screen.
```

Comment:
Not, the drawer is used to generated the svg of the diagram.

Reply:
done